src/resolvers/index.test.ts

```diff
@@ -183,41 +162,66 @@ describe('GraphQL resolvers', () => {
         },
       ],
     ])(
-      'the %s operation returns an authentication error',
+      'the %s operation returns an authentication errors',
       async (_name, op, variables: VariableValues = {}) => {
-        const result = await server.executeOperation({
+        const {
+          body: {
+            singleResult: { errors },
+          },
+        } = (await server.executeOperation({
           query: op,
           variables,
-        })
-
-        expect(result.errors).toHaveLength(1)
+        })) as SingleGraphQLResponse<any>
 
-        if (result.errors?.length) {
-          expect(result.errors[0].extensions?.code).toEqual('UNAUTHENTICATED')
-          expect(result.errors[0].message).toEqual(
-            'You must be logged in to perform this operation'
-          )
-        }
+        expect(errors).toHaveLength(1)
+        expect(errors?.[0].extensions?.code).toEqual('UNAUTHENTICATED')
+        expect(errors?.[0].message).toEqual(
+          'You must be logged in to perform this operation'
+        )
       }
     )
   })
 
   describe('while logged in', () => {
     beforeAll(async () => {
-      server = new ApolloServer({
+      // Create a new server instance
+      server = new ApolloServer<ServerContext>({
         typeDefs,
         resolvers,
-        dataSources: () => ({
-          keystoneAPI: new KeystoneAPI(),
-          weatherAPI: new WeatherAPI(),
-        }),
-        context: () => ({
-          db,
-          user: {
-            userId: newPortalUser.userId,
-          },
-        }),
       })
+      const { cache } = server
+      // Set up dataSources and mock their methods
+      keystoneAPI = new KeystoneAPI({ cache })
+      weatherAPI = new WeatherAPI({ cache })
+      personnelAPI = new PersonnelAPI({ cache })
+
+      keystoneAPI.getLatLong = jest.fn(async () => {
```

Comment:
Instead of trying to mock out these data sources separate, I'm following the example in the apollo docs and mocking their methods alongside the rest of the setup.

https://www.apollographql.com/docs/apollo-server/testing/testing/